src/components/Missions.js

```diff
@@ -0,0 +1,68 @@
+import { useSelector, useDispatch } from 'react-redux';
+import styles from '../styles/Missions.module.css';
+import { joinMission } from '../redux/missions/missionsSlice';
+
+const Missions = () => {
+  const dispatch = useDispatch();
+  const { missions, isLoading, error } = useSelector((store) => store.missions);
+  const redColored = {
+    color: '#ea0000',
+    borderColor: '#ea0000',
+    outlineColor: '#ea0000',
+  };
+
+  if (isLoading) {
+    return <p className={styles.loading}>The missions list is loading!</p>;
+  }
+  if (error) {
+    return (
+      <p className={styles.error}>
+        Something went wrong!
+        <br />
+        { error }
+      </p>
+    );
+  }
+  if (missions.length !== 0) {
+    return (
+      <table className={styles.missions}>
+        <thead>
+          <tr>
+            <th>Misson</th>
+            <th>Description</th>
+            <th>Status</th>
+            <th> </th>
+          </tr>
+        </thead>
+        <tbody>
+          {missions.map((item) => (
+            <tr key={item.mission_id}>
+              <td className={styles.name}>{item.mission_name}</td>
+              <td className={styles.description}>{item.description}</td>
+              <td>
+                {item.reserved ? (
+                  <span className={styles.is_member}>active member</span>
+                ) : (<span className={styles.not_member}>not a member</span>)}
+              </td>
+              <td>
+                <button
+                  type="button"
+                  className={styles.join_mission}
+                  style={item.reserved ? redColored : null}
+                  onClick={() => dispatch(joinMission(item.mission_id))}
+                >
+                  {item.reserved ? (
+                    'leave mission'
+                  ) : ('join mission')}
+                </button>
+              </td>
+            </tr>
+          ))}
+        </tbody>
+      </table>
+    );
+  }
+  return <p>There are no missions available.</p>;
+};
+
+export default Missions;
```

Comment:
- [ ] Kindly write tests for all your components and functions using Jest for pure functions and **_React Testing Library_** for your components. You can learn to use the **_React Testing Library_** by reading this [article](https://www.robinwieruch.de/react-testing-library/). 👍 😎

Reply:
@Gambit142 
Hi! I'd like to know why the tests are required in the review as it wasn't in the project requirements 🙂

Reply:
Hi @Zilola-Nazarova, they are required:

![image](https://github.com/Zilola-Nazarova/space-travelers-hub/assets/77106962/210c22f0-3373-4527-ae9b-394cdae22dbb)

This is the link to that part of your requirements:

https://github.com/microverseinc/curriculum-react-redux/blob/main/group-project/project_space_travelers_hub.md#:~:text=3%20members%5D_.-,Technical%20set%20up**,-Set%20up%20the

Reply:
@Gambit142 
But does the setup of the libraries mean writing tests?
Usually, they require to 'write' the tests directly (not just set up the library)

Reply:
I just saw that, thank you! 🙂 
I'm surprised I didn't see that because I re-read the requirements like 5 times 🔍 :face_with_spiral_eyes: 
Maybe because it was capitalized I thought it was the name of a section and automatically was skipping it.